controlplane/kubeadm/internal/cluster.go

```diff
@@ -206,7 +206,28 @@ func (m *ManagementCluster) RemoveEtcdMemberForMachine(ctx context.Context, clus
 		return err
 	}
 
-	return cluster.removeMemberForNode(ctx, machine.Status.NodeRef.Name)
+	controlPlaneNodes, err := cluster.getControlPlaneNodes(ctx)
+	if err != nil {
+		return err
+	}
+
+	nodeToRemove := machine.Status.NodeRef.Name
+	errs := []error{}
+	for _, node := range controlPlaneNodes.Items {
+		nodeForEtcdClient := node.Name
+		if nodeForEtcdClient == nodeToRemove {
+			continue
+		}
+
+		err = cluster.removeMemberForNode(ctx, nodeForEtcdClient, nodeToRemove)
+		if err == nil {
+			return nil
```

Comment:
Is the actual intent to return here if there is no error?

Reply:
Yeah, since we iterate over nodes except than will be removed for etcd client proxy then we need to returning soon if it's successful. we won't remove same etcd member twice right?

Reply:
let's add a comment to that effect, for posterity :)

Reply:
alright :grin: 
just added a little comment for this